input/pagecontent/smart-web-messaging.md

```diff
@@ -388,15 +410,241 @@ targetWindow.postMessage({
       // additional details as needed
     }
   }
-}, targetOrigin);
+}, ehrOrigin);
 ```
 
+The EHR then performs the requested operation, creating an object in an internal
+scratchpad and assigning it id `123` before finally responding with:
+
+```js
+// From EHR -> app
+appWindow.postMessage({
+  "messageId": "<some new uid>",
+  "responseToMessageId": "<corresponding request messageId>",
+  "payload": {
+    "status": "201 Created",
+    "location": "ServiceRequest/123"
+  }
+}, appOrigin);
+```
+
+##### `scratchpad.read`
+
+The `scratchpad.read` operation allows for selection of either a single resource
+from the scratchpad by requesting its `location` handle, or for selection of the
+*entire* contents of the scratchpad by omitting `location` from the operation
+parameter list.
+
+###### Request `payload`
+
+| Property              | Optionality | Type   | Description |
+| --------------------- | ----------- | ------ | ----------- |
+| `location`            | OPTIONAL    | string | Takes the form `ResourceType/Id` when provided.  |
```

Comment:
In our last 1:1 discussion we talked about revamping the read operation to return a single resource or array of resources instead of a Bundle; did you decide against this?

Reply:
That's right.  I wanted to first code it up to show what the simplified bundle would [look like](https://github.com/HL7/smart-web-messaging/commit/c3ac78bd5d93a13c93e5dfadb2ffdaa90f08ff48), then replace it with an [array for everything](https://github.com/HL7/smart-web-messaging/commit/2c18e57f0e54ac3e08d148e0c23233538561075c), then with the option of [an array of Resources for 'all the scratchpad' or a single Resource when specifying a location](https://github.com/HL7/smart-web-messaging/commit/5281a6486deca8a5a41dd3b3201a88c028556d0d).

Overall, I think I land somewhere between the second and third options.  I like that the awkwardness with `fullUrl` goes away with the array, and it will be convenient for developers to request a single Resource.  I'm still not sure I like having the two very similar names, though, `resource` and `resources` in the final iteration.

I considered `resource` and `scratchpad` to further distance the two attributes, but wanted to get some feedback first.

Reply:
Ah, sorry - I meant to resolve this conversation with a comment.  Here's what I meant to say before closing it.

I resolve this by renaming `resources` to `scratchpad` in [this commit](https://github.com/HL7/smart-web-messaging/commit/23ab5e26684281560882c4daa06b0631f52de79e).